src/server/pkg/storage/fileset/storage.go

```diff
@@ -269,10 +269,13 @@ func (s *Storage) compactSpec(ctx context.Context, fileSet string, compactedFile
 func (s *Storage) Delete(ctx context.Context, fileSet string) error {
 	fileSet = applyPrefix(fileSet)
 	return s.objC.Walk(ctx, fileSet, func(name string) error {
+		if err := s.objC.Delete(ctx, name); err != nil {
```

Comment:
I think we are eventually going to want this to be handled by a tombstone so we have some way to handle failing to delete the semantic reference. Also, you can just return the result of the `DeleteSemanticReference` call.